signal-desktop-bin(x86_64-linux; aarch64-linux; darwin): 7.47.0 -> 7.52.0; signal-desktop-bin: replace apple emoji sheets

Gliczy opened the pull request:
https://github.com/signalapp/Signal-Desktop/compare/v7.47.0...v7.51.0
https://github.com/signalapp/Signal-Desktop/compare/v7.51.0...v7.52.0

Signal now ships with `emoji-sheet-32.webp` and `emoji-sheet-64.webp`, both use Apple emoji.
This removes both and replace them with the equivalent noto-fonts emoji sheets.
Also, `emoji-datasource-apple` is no longer bundled, so I removed the `rm` for it.

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Gliczy committed:
signal-desktop-bin: 7.47.0 -> 7.52.0

Gliczy committed:
signal-desktop-bin(aarch64-linux): 7.47.0 ->7.52.0

Gliczy committed:
signal-desktop-bin: replace apple emoji sheets
Signal now ships with `emoji-sheet-32.webp` and `emoji-sheet-64.webp`, both use Apple emoji.
This commit removes both and replace them with the equivalent noto-fonts emoji sheets.

Gliczy commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 400945`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>signal-desktop-bin</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>signal-desktop-bin</li>
  </ul>
</details>

emilazy commented:
Thanks! `copy-noto-emoji.py` could probably be simplified to only produce the Jumbomoji paths as a result?

It might be good to have an explicit check for the version of the emoji set being used so that we don’t forget to update the Noto one.

hraban commented:
Thanks signal-desktop-bin is practically broken on macos in current nixpkgs-unstable so this is welcome.

Would you happen to know why the update script failed on this?

Gliczy committed:
signal-desktop-bin: cleanup `copy-noto-emoji.py`

Gliczy committed:
signal-desktop-bin(darwin): 7.51.0 -> 7.52.0

Gliczy commented:
Bumped to `7.52.0`, updated the `darwin` package and cleaned up `copy-noto-emoji.py`.

@emilazy I'm not sure how we should explicitly check for the emoji version, did you have something in mind?

@hraban do you mind testing the darwin build?

hraban commented:
tested it and it works 👍  aarch64-darwin
